### ISSUE TITLE
Feature Request: Enable ngram_range and max_features for transformers

### ISSUE THREAD

**Issue by HaaLeo** [amaiya/ktrain#190]
# Description
I am using the ktrain's `Transformer` interface. 
Browsing the source code I found out that the `ngram_range` and `max_features` arguments are omitted for the [`Transformer` constructor](https://github.com/amaiya/ktrain/blob/master/ktrain/text/preprocessor.py#L1041) and `ngram_range` is ignored in the [`TransformersPreprocessor`](https://github.com/amaiya/ktrain/blob/master/ktrain/text/preprocessor.py#L828).

# Proposal
I do not see an obvious reason for those parameters to be disabled. Therefore I want to propose to reveal them in the `Transformer` interface.

**Comment by amaiya** [comment]
Hi @HaaLeo,

The reason they are missing or ignored is because they aren't really applicable to pretrained Transformer-based models.   For instance, the `max_features` argument is for when the vocabulary is learned from the training set (which is what happens with models like `fasttext` and `nbsvm`).  Pretrained  Transformer models, on the other hand, have pretrained vocabularies of words and subwords, so the `max_features` argument doesn't apply.  (Transformer methods like BERT can make better sense of new, never-before-seen words in test set for these reasons.)   Similarly , the `ngram_range` feature only makes sense for "bag of words" models like `fasttext` and `nbsvm`, not `transformer` models (or LSTM/RNN/GRU for that matter).

**Comment by HaaLeo** [comment]
Hi @amaiya, thank you for the quick response. 
I agree with your comment. However, those parameter are also used in the [`preprocess_*`](https://github.com/amaiya/ktrain/blob/d80b8fdb24326899eb2c50bd62be2dbbc8d2d476/ktrain/text/preprocessor.py#L568) methods. Maybe the preprocess methods should then receive those parameters as arguments?

**Comment by amaiya** [comment]
Those parameters are not used in `preprocess_*` methods for classes associated with Transformers or BERT.  They are accepted in the constructor right now for consistency, but ignored. They are only used in `StandardTextPreprocessor`

**Comment by HaaLeo** [comment]
Thank‘s for the clarification 👍🏼
